spec/jobs/channels/whatsapp/templates_sync_scheduler_job_spec.rb

```diff
@@ -23,5 +23,23 @@
         have_been_enqueued.with(non_synced).on_queue('low')
       )
     end
+
+    it 'schedules templates_sync_job for oldest synced channels first' do
+      stub_const('Limits::BULK_EXTERNAL_HTTP_CALLS_LIMIT', 2)
+      stub_request(:post, 'https://waba.360dialog.io/v1/configs/webhook')
+      non_synced = create(:channel_whatsapp, sync_templates: false, message_templates_last_updated: nil)
+      synced_recently = create(:channel_whatsapp, sync_templates: false, message_templates_last_updated: 4.hours.ago)
+      synced_old = create(:channel_whatsapp, sync_templates: false, message_templates_last_updated: 6.hours.ago)
+      described_class.perform_now
+      expect(Channels::Whatsapp::TemplatesSyncJob).not_to(
```

Comment:
@sojan-official  `synced_recently` with 4.hours.ago should be scheduled, right?

Reply:
@vishnu-narayanan We have stubbed the limit to 2, so only two channels get priority. As per new logic, it will be the non-synced channel, and oldest channel, 4 hours being the latest, will be discarded. LMK if this makes sense

-  synced nil 
- synced 6 hours ago